File: app/src/main/java/com/konifar/materialcat/models/pojo/PhotoSource.java
package com.konifar.materialcat.models.pojo;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class PhotoSource extends Model {

    @SerializedName("photos")
    private Photos photos;
    @SerializedName("stat")
    private String stat;

    public List<Photo> getPhotos() {
        return photos.photo;
    }

    public class Photos extends Model {
        @SerializedName("page")
        private int page;
        @SerializedName("pages")
        private int pages;
        @SerializedName("perpage")
        private int perpage;
        @SerializedName("total")
        private String total;
        @SerializedName("photo")
        private List<Photo> photo;
    }

}

File: app/src/main/kotlin/com/konifar/materialcat/models/pojo/PhotoSource.kt
package com.konifar.materialcat.models.pojo

import com.google.gson.annotations.SerializedName

class PhotoSource : Model() {

    @SerializedName("photos")
    private val photos: Photos? = null
    @SerializedName("stat")
    private val stat: String? = null

    fun getPhotos(): List<Photo>? {
        return photos!!.photo
    }

    inner class Photos : Model() {
        @SerializedName("page")
        private val page: Int = 0
        @SerializedName("pages")
        private val pages: Int = 0
        @SerializedName("perpage")
        private val perpage: Int = 0
        @SerializedName("total")
        private val total: String? = null
        @SerializedName("photo")
        val photo: List<Photo>? = null
    }

}
